File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/history/ContentCopyHistoryEntry.js
define([ "stjs", "util/Exceptions", "as/dto/history/HistoryEntry" ], function(stjs, exceptions, HistoryEntry) {
	var ContentCopyHistoryEntry = function() {
		HistoryEntry.call(this);
	};
	stjs.extend(ContentCopyHistoryEntry, HistoryEntry, [ HistoryEntry ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.history.ContentCopyHistoryEntry';
		constructor.serialVersionUID = 1;
		prototype.externalCode = null;
		prototype.path = null;
		prototype.gitCommitHash = null;
		prototype.gitRepositoryId = null;
		prototype.externalDmsId = null;
		prototype.externalDmsCode = null;
		prototype.externalDmsLabel = null;
		prototype.externalDmsAddress = null;

		prototype.getExternalCode = function() {
			return this.externalCode;
		};
		prototype.setExternalCode = function(externalCode) {
			this.externalCode = externalCode;
		};
		prototype.getPath = function() {
			return this.path;
		};
		prototype.setPath = function(path) {
			this.path = path;
		};
		prototype.getGitCommitHash = function() {
			return this.gitCommitHash;
		};
		prototype.setGitCommitHash = function(gitCommitHash) {
			this.gitCommitHash = gitCommitHash;
		};
		prototype.getGitRepositoryId = function() {
			return this.gitRepositoryId;
		};
		prototype.setGitRepositoryId = function(gitRepositoryId) {
			this.gitRepositoryId = gitRepositoryId;
		};
		prototype.getExternalDmsId = function() {
			return this.externalDmsId;
		};
		prototype.setExternalDmsId = function(externalDmsId) {
			this.externalDmsId = externalDmsId;
		};
		prototype.getExternalDmsCode = function() {
			return this.externalDmsCode;
		};
		prototype.setExternalDmsCode = function(externalDmsCode) {
			this.externalDmsCode = externalDmsCode;
		};
		prototype.getExternalDmsLabel = function() {
			return this.externalDmsLabel;
		};
		prototype.setExternalDmsLabel = function(externalDmsLabel) {
			this.externalDmsLabel = externalDmsLabel;
		};
		prototype.getExternalDmsAddress = function() {
			return this.externalDmsAddress;
		};
		prototype.setExternalDmsAddress = function(externalDmsAddress) {
			this.externalDmsAddress = externalDmsAddress;
		};
}, {});
	return ContentCopyHistoryEntry;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/history/ContentCopyHistoryEntry.java
/*
 * Copyright 2015 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.history;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.history.ContentCopyHistoryEntry")
public class ContentCopyHistoryEntry extends HistoryEntry
{

    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String externalCode;

    @JsonProperty
    private String path;

    @JsonProperty
    private String gitCommitHash;

    @JsonProperty
    private String gitRepositoryId;

    @JsonProperty
    private Long externalDmsId;

    @JsonProperty
    private String externalDmsCode;

    @JsonProperty
    private String externalDmsLabel;

    @JsonProperty
    private String externalDmsAddress;


    @JsonIgnore
    public String getExternalCode()
    {
        return externalCode;
    }

    @JsonIgnore
    public void setExternalCode(String externalCode)
    {
        this.externalCode = externalCode;
    }

    @JsonIgnore
    public String getPath()
    {
        return path;
    }

    @JsonIgnore
    public void setPath(String path)
    {
        this.path = path;
    }

    @JsonIgnore
    public String getGitCommitHash()
    {
        return gitCommitHash;
    }

    @JsonIgnore
    public void setGitCommitHash(String gitCommitHash)
    {
        this.gitCommitHash = gitCommitHash;
    }

    @JsonIgnore
    public String getGitRepositoryId()
    {
        return gitRepositoryId;
    }

    @JsonIgnore
    public void setGitRepositoryId(String gitRepositoryId)
    {
        this.gitRepositoryId = gitRepositoryId;
    }

    @JsonIgnore
    public Long getExternalDmsId()
    {
        return externalDmsId;
    }

    @JsonIgnore
    public void setExternalDmsId(Long externalDmsId)
    {
        this.externalDmsId = externalDmsId;
    }

    @JsonIgnore
    public String getExternalDmsCode()
    {
        return externalDmsCode;
    }

    @JsonIgnore
    public void setExternalDmsCode(String externalDmsCode)
    {
        this.externalDmsCode = externalDmsCode;
    }

    @JsonIgnore
    public String getExternalDmsLabel()
    {
        return externalDmsLabel;
    }

    @JsonIgnore
    public void setExternalDmsLabel(String externalDmsLabel)
    {
        this.externalDmsLabel = externalDmsLabel;
    }

    @JsonIgnore
    public String getExternalDmsAddress()
    {
        return externalDmsAddress;
    }

    @JsonIgnore
    public void setExternalDmsAddress(String externalDmsAddress)
    {
        this.externalDmsAddress = externalDmsAddress;
    }

    @JsonIgnore
    public static long getSerialversionuid()
    {
        return serialVersionUID;
    }

}
